### PR TITLE
Pull request/5eeadc76

### PR DESCRIPTION
**Pull request by phaverty**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,3 @@
 julia 0.2-
-Compat 0.2.9
```
Comment (carlobaldassi) [inline REQUIRE]: Why removing `Compat`?

File: src/ArgParse.jl [hunk 2]
```diff
@@ -1,10 +1,9 @@
-isdefined(:OptionsMod) || require("Options")
+isdefined(:OptionsMod) || @eval import Options
```
Comment (carlobaldassi) [inline src/ArgParse.jl]: I don't think `@eval` is needed here, is it?
Also, I'm not on latest julia master: does this still print a warning?

Comment (phaverty) [inline src/ArgParse.jl]: Hi,

Thanks for considering this pull request.
For the @eval part, I'm copying the suggestion in
https://github.com/JuliaLang/julia/issues/12233
This change gets rid of the warning about `require` being deprecated.
However the warning "Warning: requiring "Options" did not define a
corresponding module." remains.
Is Options still necessary? The README for that package states that much of
its functionality is now in base.

The removal of Compat.jl was unrelated. Sorry, that should have been a
separate commit at least. It looks like it is no longer necessary? I was
hoping that removing a dependency would get the load time down a bit to
make command line programs a bit snappier. BTW, I was also trying out the
new precompilation feature to get a really fast startup of this package,
but something about that makes julia fail to find Options.OptionsMod.

Regards,

Pete

____________________
Peter M. Haverty, Ph.D.
Genentech, Inc.
phaverty@gene.com

Comment (carlobaldassi) [inline src/ArgParse.jl]: Ok it's probably safer to use `@eval` (actually in https://github.com/JuliaLang/julia/issues/12233 Jeff later suggested `eval(Main, :(import Options))`)  even though I'm not sure exacly what would be different in this case.

About Options being necessary, see https://github.com/carlobaldassi/ArgParse.jl/issues/13

The new warning is indeed annoying, this issue is https://github.com/JuliaLang/Options.jl/issues/4.

Compat is needed for `startswith` and other changes, see e.g. https://github.com/carlobaldassi/ArgParse.jl/commit/996d7be91142731edc07ab10eead324201cf936f

Comment (phaverty) [inline src/ArgParse.jl]: Thanks for the links to those conversations. I put in a cleaner pull
request with just the @eval change, if you are interested.

Regards,

Pete

____________________
Peter M. Haverty, Ph.D.
Genentech, Inc.
phaverty@gene.com

**Pull request closed without merging**